### PR TITLE
gh-120811: Fix reference leak upon `_PyContext_Exit` failure

### PR DESCRIPTION
**Pull request by ZeroIntensity**
* Issue: gh-120811

### CODE REVIEW

**Review by Eclips4 — APPROVED** [review]

**Comment by AraHaan** [comment]
can this be backported to the 3.11 branch but require people who want the fix to manually compile themselves?

**Comment by Eclips4** [comment]
> can this be backported to the 3.11 branch but require people who want the fix to manually compile themselves as well?

Unfortunately, no. 3.11 branch accepts now only a security fixes.

**Review by kumaraditya303 — CHANGES_REQUESTED** [review]
Please add a news entry, it is a user facing bug not internal.

**Comment by ZeroIntensity** [comment]
I have made the requested changes; please review again

File: Misc/NEWS.d/next/Library/2024-06-21-14-32-56.gh-issue-120811.eBmVTV.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix memory leak in ``contextvars.Context.run()``
```
Comment (kumaraditya303) [inline Misc/NEWS.d/next/Library/2024-06-21-14-32-56.gh-issue-120811.eBmVTV.rst]:
```suggestion
Fix possible memory leak in :meth:`contextvars.Context.run`.
```

**Review by kumaraditya303 — APPROVED** [review]

**Pull request merged**